File: skills/harmonyos/development/hmos-one-sdk-skill/hmos-sdk-basic-skill/Account Kit(华为账号服务)/hmos-account-kit-get-realname-age/assets/example_server.ts
import axios from 'axios';

const HUAWEI_TOKEN_URL = 'https://oauth-login.cloud.huawei.com/oauth2/v3/token';
const HUAWEI_AGEGANGE_URL = 'https://account-api.huawei.com/v1/user/realnameAgeRange';

interface TokenResponse {
  access_token: string;
  refresh_token: string;
  expires_in: number;
}

interface AgeRangeResponse {
  ageRange: string;
}

export class RealNameAgeRangeServer {
  private clientId: string;
  private clientSecret: string;

  constructor(clientId: string, clientSecret: string) {
    this.clientId = clientId;
    this.clientSecret = clientSecret;
  }

  async getAccessToken(authorizationCode: string): Promise<TokenResponse> {
    try {
      const response = await axios.post(HUAWEI_TOKEN_URL, {
        grant_type: 'authorization_code',
        code: authorizationCode,
        client_id: this.clientId,
        client_secret: this.clientSecret
      }, {
        headers: { 'Content-Type': 'application/json' }
      });

      const data = response.data;
      console.log(`Access Token obtained, expires in ${data.expires_in} seconds`);
      
      return {
        access_token: data.access_token,
        refresh_token: data.refresh_token,
        expires_in: data.expires_in
      };
    } catch (error) {
      console.error('Failed to get Access Token:', error);
      throw error;
    }
  }

  async getRealNameAgeRange(accessToken: string): Promise<string> {
    try {
      const response = await axios.get(HUAWEI_AGEGANGE_URL, {
        headers: {
          'Authorization': `Bearer ${accessToken}`,
          'Content-Type': 'application/json'
        }
      });

      const data: AgeRangeResponse = response.data;
      console.log(`Age Range obtained: ${data.ageRange}`);
      
      return data.ageRange;
    } catch (error) {
      console.error('Failed to get Age Range:', error);
      throw error;
    }
  }

  async refreshAccessToken(refreshToken: string): Promise<string> {
    try {
      const response = await axios.post(HUAWEI_TOKEN_URL, {
        grant_type: 'refresh_token',
        refresh_token: refreshToken,
        client_id: this.clientId,
        client_secret: this.clientSecret
      }, {
        headers: { 'Content-Type': 'application/json' }
      });

      const data = response.data;
      console.log(`Access Token refreshed, expires in ${data.expires_in} seconds`);
      
      return data.access_token;
    } catch (error) {
      console.error('Failed to refresh Access Token:', error);
      throw error;
    }
  }

  async processAuthorizationCode(authorizationCode: string): Promise<string> {
    try {
      const tokenResponse = await this.getAccessToken(authorizationCode);
      const ageRange = await this.getRealNameAgeRange(tokenResponse.access_token);
      
      return ageRange;
    } catch (error) {
      console.error('Failed to process Authorization Code:', error);
      throw error;
    }
  }
}

export async function main(authorizationCode: string): Promise<void> {
  const clientId = 'YOUR_CLIENT_ID';
  const clientSecret = 'YOUR_CLIENT_SECRET';
  
  const server = new RealNameAgeRangeServer(clientId, clientSecret);
  
  try {
    const ageRange = await server.processAuthorizationCode(authorizationCode);
    console.log(`Final Age Range: ${ageRange}`);
  } catch (error) {
    console.error('Error in main process:', error);
  }
}

File: skills/harmonyos/development/hmos-one-sdk-skill/hmos-sdk-basic-skill/Account Kit(华为账号服务)/hmos-account-kit-quick-phonenumber-verification/assets/example_server.java
import com.alibaba.fastjson2.JSONObject;
import org.apache.http.NameValuePair;
import org.apache.http.client.entity.UrlEncodedFormEntity;
import org.apache.http.client.methods.HttpPost;
import org.apache.http.impl.client.CloseableHttpClient;
import org.apache.http.impl.client.HttpClients;
import org.apache.http.message.BasicNameValuePair;
import org.apache.http.util.EntityUtils;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class QuickPhoneNumberServer {
    
    private static final String TOKEN_URL = "https://oauth-login.cloud.huawei.com/oauth2/v3/token";
    private static final String USER_INFO_URL = "https://account.cloud.huawei.com/rest.php?nsp_svc=GOpen.User.getInfo";
    
    private String clientId;
    private String clientSecret;
    
    public QuickPhoneNumberServer(String clientId, String clientSecret) {
        this.clientId = clientId;
        this.clientSecret = clientSecret;
    }
    
    public AccessTokenResult getAccessTokenByCode(String authorizationCode) throws IOException {
        CloseableHttpClient httpClient = HttpClients.createDefault();
        HttpPost httpPost = new HttpPost(TOKEN_URL);
        
        List<NameValuePair> params = new ArrayList<>();
        params.add(new BasicNameValuePair("grant_type", "authorization_code"));
        params.add(new BasicNameValuePair("code", authorizationCode));
        params.add(new BasicNameValuePair("client_id", clientId));
        params.add(new BasicNameValuePair("client_secret", clientSecret));
        params.add(new BasicNameValuePair("supportAlg", "PS256"));
        
        httpPost.setHeader("Content-Type", "application/x-www-form-urlencoded");
        httpPost.setEntity(new UrlEncodedFormEntity(params));
        
        String responseBody = EntityUtils.toString(httpClient.execute(httpPost).getEntity());
        JSONObject result = JSONObject.parseObject(responseBody);
        
        AccessTokenResult tokenResult = new AccessTokenResult();
        tokenResult.accessToken = result.getString("access_token");
        tokenResult.refreshToken = result.getString("refresh_token");
        tokenResult.expiresIn = result.getInteger("expires_in");
        tokenResult.idToken = result.getString("id_token");
        tokenResult.scope = result.getString("scope");
        tokenResult.tokenType = result.getString("token_type");
        
        httpClient.close();
        return tokenResult;
    }
    
    public PhoneNumberResult getUserPhoneNumber(String accessToken) throws IOException {
        CloseableHttpClient httpClient = HttpClients.createDefault();
        HttpPost httpPost = new HttpPost(USER_INFO_URL);
        
        List<NameValuePair> params = new ArrayList<>();
        params.add(new BasicNameValuePair("access_token", accessToken));
        
        httpPost.setHeader("Content-Type", "application/x-www-form-urlencoded");
        httpPost.setEntity(new UrlEncodedFormEntity(params));
        
        String responseBody = EntityUtils.toString(httpClient.execute(httpPost).getEntity());
        JSONObject result = JSONObject.parseObject(responseBody);
        
        PhoneNumberResult phoneResult = new PhoneNumberResult();
        phoneResult.openID = result.getString("openID");
        phoneResult.unionID = result.getString("unionID");
        phoneResult.mobileNumber = result.getString("mobileNumber");
        phoneResult.purePhoneNumber = result.getString("purePhoneNumber");
        phoneResult.phoneCountryCode = result.getString("phoneCountryCode");
        
        httpClient.close();
        return phoneResult;
    }
    
    public AccessTokenResult refreshAccessToken(String refreshToken) throws IOException {
        CloseableHttpClient httpClient = HttpClients.createDefault();
        HttpPost httpPost = new HttpPost(TOKEN_URL);
        
        List<NameValuePair> params = new ArrayList<>();
        params.add(new BasicNameValuePair("grant_type", "refresh_token"));
        params.add(new BasicNameValuePair("refresh_token", refreshToken));
        params.add(new BasicNameValuePair("client_id", clientId));
        params.add(new BasicNameValuePair("client_secret", clientSecret));
        
        httpPost.setHeader("Content-Type", "application/x-www-form-urlencoded");
        httpPost.setEntity(new UrlEncodedFormEntity(params));
        
        String responseBody = EntityUtils.toString(httpClient.execute(httpPost).getEntity());
        JSONObject result = JSONObject.parseObject(responseBody);
        
        AccessTokenResult tokenResult = new AccessTokenResult();
        tokenResult.accessToken = result.getString("access_token");
        tokenResult.refreshToken = result.getString("refresh_token");
        tokenResult.expiresIn = result.getInteger("expires_in");
        tokenResult.tokenType = result.getString("token_type");
        
        httpClient.close();
        return tokenResult;
    }
    
    public PhoneNumberResult getPhoneNumberByAuthCode(String authorizationCode) throws IOException {
        AccessTokenResult tokenResult = getAccessTokenByCode(authorizationCode);
        return getUserPhoneNumber(tokenResult.accessToken);
    }
    
    public static class AccessTokenResult {
        public String accessToken;
        public String refreshToken;
        public Integer expiresIn;
        public String idToken;
        public String scope;
        public String tokenType;
    }
    
    public static class PhoneNumberResult {
        public String openID;
        public String unionID;
        public String mobileNumber;
        public String purePhoneNumber;
        public String phoneCountryCode;
    }
    
    public static void main(String[] args) {
        String clientId = "<Your Client ID>";
        String clientSecret = "<Your Client Secret>";
        String authorizationCode = "<Authorization Code from Client>";
        
        QuickPhoneNumberServer server = new QuickPhoneNumberServer(clientId, clientSecret);
        
        try {
            PhoneNumberResult result = server.getPhoneNumberByAuthCode(authorizationCode);
            
            System.out.println("=== Phone Number Retrieval Result ===");
            System.out.println("OpenID: " + result.openID);
            System.out.println("UnionID: " + result.unionID);
            System.out.println("Mobile Number: " + result.mobileNumber);
            System.out.println("Pure Phone Number: " + result.purePhoneNumber);
            System.out.println("Phone Country Code: " + result.phoneCountryCode);
            
        } catch (IOException e) {
            System.err.println("Failed to get phone number: " + e.getMessage());
            e.printStackTrace();
        }
    }
}